File: src/main/java/com/project/GiveTest/services/QuizService.java
package com.project.GiveTest.services;

import com.project.GiveTest.entities.exam.Category;
import com.project.GiveTest.entities.exam.Quiz;
import org.springframework.http.ResponseEntity;

import java.util.List;
import java.util.Set;

public interface QuizService {

    public Quiz addQuiz(Quiz quiz);

    public Quiz updateQuiz(Quiz quiz);

    public Set<Quiz> getQuizzes();

    public Quiz getQuiz(Long qId);

    public void deleteQuiz(Long qId);

    public List<Quiz> getQuizzesOfCategory(Category category);

    public List<Quiz> getActiveQuizzes();
    public List<Quiz> getActiveQuizzesOfCategory(Category c);
}

File: src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private Http:HttpClient) { }


  public Quizzes(){
    return this.Http.get(`${baseUrl}/quiz/`)
  }

  public addCategory(quiz:any){
    return this.Http.post(`${baseUrl}/quiz/`,quiz)
  }
  public deleteQuiz(qId:any){
    return this.Http.delete(`${baseUrl}/quiz/${qId}`);
  }

  public updateQuiz(quiz:any){
    return this.Http.put(`${baseUrl}/quiz/`,quiz);
  }

  public getQuiz(qId:any){
    return this.Http.get(`${baseUrl}/quiz/${qId}`);
  }

  public getQuizzesOfCategory(cid:any){
    return this.Http.get(`${baseUrl}/quiz/category/${cid}`);
  }

  public getActiveQuizzes(){
    return this.Http.get(`${baseUrl}/quiz/active`)
  }

  public getActiveQuizzesOfCategory(cid:any){
    return this.Http.get(`${baseUrl}/quiz/category/active/${cid}`);
  }
}
